internal/controllers/common/resource_definitions/resource_definitions.go

```diff
@@ -976,6 +981,27 @@ func NewConsoleLink(cr *operatorv1beta1.Cryostat, url string) *consolev1.Console
 	}
 }
 
+func SetAdditionalCorsAllowedOrigins(specs *ServiceSpecs) (*configv1.APIServer, error) {
+	kubeconfig := os.Getenv(("KUBECONFIG"))
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+
+	configClient := configv1client.NewForConfigOrDie(config)
+
+	apiServer, err := configClient.APIServers().Get(context.Background(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	apiServer.Spec.AdditionalCORSAllowedOrigins = append(
+		apiServer.Spec.AdditionalCORSAllowedOrigins,
+		specs.CoreURL.Hostname(),
+	)
+
+	return configClient.APIServers().Update(context.Background(), apiServer, metav1.UpdateOptions{})
```

Comment:
I think you should be able to use the existing client from the Cryostat controller to get and update the APIServer object. You just need to register the config/v1 API with the Manager's Scheme here:
https://github.com/cryostatio/cryostat-operator/blob/5c399ff843136198000a1bf0b99837d0c307a744/internal/main.go#L56-L59

Then you should be able to use the APIServer type just like any other built-in Kubernetes resource.

Reply:
I think it also makes sense to move this functionality to cryostat_controller.go, since you'll have the client object already there.

Reply:
`reqLogger.Info()` should send output to the logs in `oc logs cryostat-operator-controller-manager-78c899f994-9kt9m`, right?
 
The tests seem to notice an error thrown at `r.Client.Get(context.Background(), types.NamespacedName{Name: "cluster"}, apiServer)`, except when I write `reqLogger.Error(err, "Failed to get APIServer config")`, I don't see any errors printed in the logs. Any ideas?

Reply:
I'm seeing the following error when building and deploying your PR branch:
`E0106 19:09:30.129907       1 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156: Failed to watch *v1.APIServer: failed to list *v1.APIServer: apiservers.config.openshift.io is forbidden: User "system:serviceaccount:cryostat-operator-system:cryostat-operator-service-account" cannot list resource "apiservers" in API group "config.openshift.io" at the cluster scope
`

Try it out with: `make deploy IMG=quay.io/ebaron/cryostat-operator:oauth-logout-01`.

You'll need to give get/list/watch/update permissions to the operator with the kubebuilder marker comments here:
https://github.com/cryostatio/cryostat-operator/blob/91179772f4277810d4d82f4f21f0c5c57acf7c43/internal/controllers/cryostat_controller.go#L104-L121

Then run `make manifests`.

Reply:
> Try it out with: make deploy IMG=quay.io/ebaron/cryostat-operator:oauth-logout-01.

Thanks, I can see the permissions error now.

When I ran the following after adding `// +kubebuilder:rbac:groups=config.openshift.io,resources=apiservers,verbs=list;update
`, I see an unrelated permissions error `User \"system:serviceaccount:default:cryostat-operator-service-account\" cannot delete resource \"clusterrolebindings\" in API group \"rbac.authorization.k8s.io\" at the cluster scope"}`:

```
make generate manifests
BUILDAH_FORMAT=docker podman build -t quay.io/jalaw/cryostat-operator:latest .
podman push quay.io/jalaw/cryostat-operator:latest
DEPLOY_NAMESPACE=$(oc project -q) make deploy CORE_IMG="quay.io/jalaw/cryostat:latest" IMG="quay.io/jalaw/cryostat-operator:latest"
```

The permissions to modify `clusterrolebindings` are unchanged: `// +kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=clusterrolebindings,verbs=create;get;list;update;watch;delete`.

I think I'm missing something. Are these the right commands to build and deploy the operator?

Reply:
Check the generated config/rbac/role.yaml to see if it added the permissions correctly and didn't remove anything it shouldn't. It looks like the following for me:

```diff
diff --git a/config/rbac/role.yaml b/config/rbac/role.yaml
index 49a7f76..16b1166 100644
--- a/config/rbac/role.yaml
+++ b/config/rbac/role.yaml
@@ -26,6 +26,13 @@ rules:
   - selfsubjectaccessreviews
   verbs:
   - create
+- apiGroups:
+  - config.openshift.io
+  resources:
+  - apiservers
+  verbs:
+  - list
+  - update
 - apiGroups:
   - console.openshift.io
   resources:
```

Testing with this marker added:
`// +kubebuilder:rbac:groups=config.openshift.io,resources=apiservers,verbs=list;update`

I see the following error in the logs:
`E0106 20:43:18.250371       1 reflector.go:127] pkg/mod/k8s.io/client-go@v0.19.2/tools/cache/reflector.go:156: Failed to watch *v1.APIServer: unknown (get apiservers.config.openshift.io)`

In addition to `list` and `update`, be sure to add `get` and `watch`.

Reply:
Hmm, `config/rbac/role.yaml` looks okay. I'm still getting the same `cannot delete resource \"clusterrolebindings\"` error.
```
diff --git a/config/rbac/role.yaml b/config/rbac/role.yaml
index 49a7f76..7b35ffe 100644
--- a/config/rbac/role.yaml
+++ b/config/rbac/role.yaml
@@ -26,6 +26,15 @@ rules:
   - selfsubjectaccessreviews
   verbs:
   - create
+- apiGroups:
+  - config.openshift.io
+  resources:
+  - apiservers
+  verbs:
+  - get
+  - list
+  - update
+  - watch
 - apiGroups:
   - console.openshift.io
   resources:
```

Reply:
That's weird, it's working for me now. The only time the operator will try to delete a cluster role binding will be when a Cryostat CR is deleted. Does `make destroy_cryostat_cr` hang for you? If not, does using my build change anything: `quay.io/ebaron/cryostat-operator:oauth-logout-02`?

Reply:
> Does make destroy_cryostat_cr hang for you?

No, I can delete the cryostat CR without problems.
```
kubectl delete -f config/samples/operator_v1beta1_cryostat.yaml
cryostat.operator.cryostat.io "cryostat-sample" deleted
```

> If not, does using my build change anything: quay.io/ebaron/cryostat-operator:oauth-logout-02?

Using your build works well. I'm not sure what could be different between my build and yours?

Reply:
I'm not sure. I modified the Makefile to remove the `test-envtest` requirement from the `oci-build` target and built with `make oci-build IMG=quay.io/ebaron/cryostat-operator:oauth-logout-02`, followed by pushing to Quay with podman. Is this similar to how you're building?

Reply:
What image tag are you using for your development builds?

Reply:
Ah, I think it's my unstable wifi/vpn connection. I've been using the same build steps as you, but this time I deleted all of my pods and tried again. Something must have been hanging from the last time my vpn disconnected, even thought it said the cryostat CR pod was deleted. Thanks!

I still need to add a function to remove the cryostat origin url once the cryostat CR is deleted. When that's working, I'll open the PR.

Reply:
Sounds good!